wireapp/wire-android
# feat: add query profiling and cached paginated list [WPB-14826]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14826" title="WPB-14826" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-14826</a>  [Android] Conversation list loading very slow after closing a conversation
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Paginated conversation list is loading slowly when navigating back from conversation screen.

### Solutions

Implement caching for the `PagingData` flow - there's a ready to use extension function `cachedIn`.
Add a logic to enable or disable database profiling to have logs with query execution times if the user enabled logging (only for cipher db) - thanks to that we will be able to confirm or eliminate some queries as potential causes of slow loading of conversation list.

### Dependencies (Optional)

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/3163

### Testing

#### Test Coverage (Optional)


#### How to Test

Open conversation and navigate back - list should be cached and appear faster on the screen.
Enable logging in the app and check logcat or logs collected in the file to make sure there are also logs with query execution times.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
